database/ffldb/db.go

```diff
@@ -1669,6 +1669,94 @@ func (tx *transaction) writePendingAndCommit() error {
 	return tx.db.cache.commitTx(tx)
 }
 
+// PruneBlocks deletes the block files until it reaches the target size
+// (specificed in bytes).  Throws an error if the target size is below
+// the maximum size of a single block file.
+//
+// This function is part of the database.Tx interface implementation.
+func (tx *transaction) PruneBlocks(targetSize uint32) ([][]byte, error) {
+	// Ensure transaction state is valid.
+	if err := tx.checkClosed(); err != nil {
+		return nil, err
+	}
+
+	// Ensure the transaction is writable.
+	if !tx.writable {
+		str := "prune blocks requires a writable database transaction"
+		return nil, makeDbErr(database.ErrTxNotWritable, str, nil)
+	}
+
+	if targetSize < tx.db.store.maxBlockFileSize {
```

Comment:
not 288*blocksize?

Reply:
No because I wanted the pruning behavior to be similar to bitcoin core's where the user can set a prune size. For example, the user could want a prune size of 70GB instead of the smallest possible size.

Reply:
That's true that we have to keep at least 2 days worth of blocks but that's not the check being done here. The check here is if the database can actually reach the `targetSize` that the caller asked for. Since each of the `.blk` files are 512MiB each, it's impossible to reach anything less than that. For example, if the caller asks for a target size of  256MiB, it'll error out here.

That's the only guarantee that the database makes. The caller should be careful to not ask for anything lower than what it'd need. That's enforced in `config.go`